src/Core/Managed/Shared/Extensibility/TelemetryConfiguration.cs

```diff
@@ -214,31 +206,40 @@ internal set
         }
 
         /// <summary>
-        /// Gets or sets the telemetry channel.
+        /// Gets or sets the telemetry channel for the default sink.
         /// </summary>
         public ITelemetryChannel TelemetryChannel
         {
             get
             {
-                return this.telemetryChannel;
+                // We do not ensure not disposed here because TelemetryChannel is accessed during configuration disposal.
+                return this.telemetrySinks[DefaultSinkIndex].TelemetryChannel;
             }
 
             set
             {
-                ITelemetryChannel oldChannel = this.telemetryChannel;
-                this.telemetryChannel = value;
+                this.EnsureNotDisposed();
```

Comment:
we typically wouldn't throw in situations like this to not crash customer code. Just return nothing.